### ISSUE TITLE
Memory leak with CartesianChart

### ISSUE THREAD

**Issue by fablacky** [Live-Charts/Live-Charts#936]
#### How to reproduce?

Just add a <lvc: CartesianChart /> tag to any xaml. Loading and unloading the usercontrol containing said tag will show many instances being created instead of a single one being disposed and recreated.

#### I am using

<table>
<tr>
    <td>Live-Charts version</td>
    <td>0.9.7</td>
</tr>
<tr>
    <td>.Net Version</td>
    <td>4.6.1</td>
</tr>
<tr>
    <td>Windows</td>
    <td>10</td>
</tr>
</table>

**Comment by MatjazJost** [comment]
Any solution already for this memory leak

**Comment by TransportUnit** [comment]
I might be a little late to this conversation, but after facing the same problem, I ran a memory profiler on a small sample program where I'd repeatedly load and unload a CartesianChart.

As it turns out, the property (LiveCharts.Wpf.Charts.Base.Chart.)ChartLegend causes the memory leak. Or, to be more precise, its default value (DefaultLegend). I haven't actually found out why this might be the case, but as I am not actually in need of a legend in my chart, I just set it to null after its initialisation:

```csharp

LiveCharts.Wpf.CartesianChart chart = new LiveCharts.Wpf.CartesianChart();

chart.ChartLegend = null;

```

This fixes the problem (at least for me). I'm happy if this still helps anyone.

**Comment by mjanulaitis** [comment]
@TransportUnit stepping through the code above I can verify the timer is stopped however according to .Net Memory Profiler there is a dispatch timer that is still leaking.  In my case each opening and closing causes the ChartUpdater to leak over 700,000 bytes.  That's huge!

**Comment by mjanulaitis** [comment]
Using Task Manager I can watch this control crush memory.  It leaks like a funnel.  It seems this project is dead in the water with no updates for several years and 30 outstanding pull requests.  So sad this is not production ready :(

**Comment by TransportUnit** [comment]
@mjanulaitis Does your problem only apply to the CartesianChart or are you using another type?
Unfortunately, I am still struggling to gain control as well. What I'd recommend is to clone the current LiveCharts repository and implement the necessary changes yourself, and to then implement it into your project. This way, you at least don't have to use any workarounds as in the code above.

I do, however, share your concerns about this project. But you also have to appreciate the work that has already been put into this project, and you have to remember that it's completely free. Also, there aren't many comparable charting packages available right now (at least for free).

As far as i know, most of this project has been written by only one person, and that's quite an achievement. I can well imagine that he does not have time or motivation left to maintain his code.
Additionally, in my opinion, those problems originate from the WPF framework itself and are just a symptom of its impracticality rather than this package to be the cause.

**Comment by mjanulaitis** [comment]
@TransportUnit Sorry for the late reply. Yes I cloned the project directly. I am only using the CartesianChart.  Yes is looks like great work which I why I have included it in my project.  I am grateful to have access to the source.

**Comment by SingletonSean** [comment]
@mjanulaitis Were you able to find a solution to the issue? Trying to not lose my job. 😄

**Comment by mjanulaitis** [comment]
@SingletonSean I did another go at it pretty good but found nothing.  The tool I was using didn't seem to produce accurate information.  Although I can open and close the charts and see the memory just keep getting larger.  Since the tool was so inaccurate I can't say what is leaking the memory now.  Maybe another memory leak detection tool will do a better job identifying what is leaking and why?

**Comment by SingletonSean** [comment]
@mjanulaitis Makes sense, I dug through the code for a couple hours and couldn't find anything interesting. I did notice that if I do not set the Series property on the chart and just display Axes and Labels, the chart gets cleaned up. This leads me to think it is something with the Series property, more specifically perhaps the SeriesCollection that the Series property binds to. This is a reactive/observable data structure, so there is a chance an object is subscribing to the SeriesCollection and never unsubscribing. That wouldn't really make sense though, the SeriesCollection should still cleanup no matter who is subscribed to it. Maybe there's some kind of pub/sub somewhere else. Have to keep looking some other day.

Anyways, I like the way you sum it up, hahaha:
> It leaks like a funnel. It seems this project is dead in the water...
